### ISSUE TITLE
Campaign just run when I am logged

### ISSUE THREAD

**Issue by sbolinches** [mautic/mautic#7474]
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description

I have verified that certain tasks are not performed automatically as it was before the update.

In particular, when a lead fills out a form, and a campaign is activated, this only happens if I'm with Mautic open on the server (and only if I'm not using Mautic in private navigation).

It is possible that it is related to other errors in which the campaigns do not send emails, etc, since that was my first indication that something was wrong. Afterwards, it happened that sometimes it worked and sometimes it did not. Finally I realized that if I perform a conversion test (through a form) if I'm logged in, or if I'm not, the result of MAutic is different (when I'm logged in, the campaign goes through normally.) If I'm not logged in, I observe in the client's history, which has been added to the campaign, but does not receive the email and the actions do not happen.

| Q   | A
| --- | ---
| Mautic version | 2.15.1
| PHP version | 7.0.22
| Browser | 

### Steps to reproduce
1. create a form
2. Create a segment
3. create a campaign
4. Create an email to sent auto when the campaign starts

### Two scene

| Nobody is using Mautic | --- | State
| --- | --- | ---
| Form is sent and lead added like contact | --- | Correct
| Form sent an email with results | --- |Correct
| Form add the lead to segment | --- | Correct
| Campaign is assigned to lead | --- | Correct
| Campaign sent mail to lead | --- | NO
| Campaign: If lead check link is assigned a Tag | --- | NO
| Campaign. Lead is assigned to other segment | --- | NO

| I am logged in Mautic | --- | State
| --- | --- | ---
| Form is sent and lead added like contact | --- | Correct
| Form sent an email with results | --- |Correct
| Form add the lead to segment | --- | Correct
| Campaign is assigned to lead | --- | Correct
| Campaign sent mail to lead | --- | Correct
| Campaign: If lead check link is assigned a Tag | --- | Correct
| Campaign. Lead is assigned to other segment | --- | Correct

It is important to say that when I use the preview of the forms, the campaigns always work (because I'm logged in) 

### My CronTask 
I do not receive any errors from cron tasks. I asked the server provider if it was a problem with permissions.

```
0,15,30,45 * * * * /usr/bin/php -c /home/dominio.es/php.ini -q /home/dominio.es/web/mautic/app/console mautic:segments:update --force --batch-limit=300

5,20,35,50 * * * * /usr/bin/php -c /home/dominio.es/php.ini -q /home/dominio.es/web/mautic/app/console mautic:campaigns:update 

10,25,40,55 * * * * /usr/bin/php -c /home/dominio.es/php.ini -q /home/dominio.es/web/mautic/app/console mautic:campaigns:trigger --force --batch-limit=100

7,17,27,37,47,57 * * * * /usr/bin/php -c /home/dominio.es/php.ini -q /home/dominio.es/web/mautic/app/console mautic:campaigns:rebuild --force --batch-limit=200

0.30 * * * * /usr/bin/php -c /home/dominio.es/php.ini -q /home/dominio.es/web/mautic/app/console mautic:messages:send

0.30 * * * * /usr/bin/php -c /home/dominio.es/php.ini -q /home/dominio.es/web/mautic/app/console mautic:emails:send

 0,30 * * * *    /usr/bin/php -c /home/dominio.es/php.ini -q  /home/dominio.es/web/mautic/app/console mautic:email:fetch

 0,30 * * * *    /usr/bin/php -c /home/dominio.es/php.ini -q  /home/dominio.es/web/mautic/app/console mautic:social:monitoring

 * 2 10 * *      /usr/bin/php -q /usr/home/dominio.es/web/mautic/update-maxmind-database >/dev/null 2>&1

```
### Log errors

**/web/mautic/app/logs**

`mautic.NOTICE: Symfony\Component\Console\Exception\RuntimeException: Aborted (uncaught exception) at /usr/home/dominio.es/web/mautic/vendor/symfony/console/Helper/QuestionHelper.php line 138 while running console command `doctrine:migrations:migrate` [] []
`
**php_errors.log**

`PHP Warning:  fgets() expects parameter 1 to be resource, string given in /usr/home/dominio.es/web/mautic/vendor/symfony/console/Helper/DialogHelper.php on line 131
`
`PHP Fatal error:  Uncaught Error: Call to a member function getParameter() on null in /usr/home/dominio.es/web/mautic/app/config/paths_helper.php:13
`
`PHP Warning:  Error while sending QUERY packet. PID=9377 in /usr/home/dominio.es/web/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php on line 105
`
`[error] [client 190.163.33.198] FastCGI: incomplete headers (0 bytes) received from server "/fcgi-bin-php70-fpm-ssl-dominio.es"
`
### Executing console doctrine:migration:status

 == Configuration

    >> Name:                                               Mautic Migrations
    >> Database Driver:                                    pdo_mysql
    >> Database Name:                                      mauticsbm
    >> Configuration Source:                               manually configured
    >> Version Table Name:                                 maumigrations
    >> Version Column Name:                                version
    >> Migrations Namespace:                               Mautic\Migrations
    >> Migrations Directory:                               /usr/home/dominio.es/web/mautic/app/migrations
    >> Previous Version:                                   2018-11-29 10:54:19 (20181129105419)
    >> Current Version:                                    2018-12-04 00:00:00 (20181204000000)
    >> Next Version:                                       Already at latest version
    >> Latest Version:                                     2018-12-04 00:00:00 (20181204000000)
    >> Executed Migrations:                                133
    >> Executed Unavailable Migrations:                    0
    >> Available Migrations:                               133
    >> New Migrations:                                     0

### Executing console doctrine:migrations:migrate

```
Warning: fgets() expects parameter 1 to be resource, string given in /usr/home/dominio.es/web/mautic/vendor/symfony/console/Helper/QuestionHelper.php on line 136
                                                         
                    Mautic Migrations                    
                                                         

WARNING! You are about to execute a database migration that could result in schema changes and data lost. Are you sure you wish to continue? (y/n)
                                                          
  [Symfony\Component\Console\Exception\RuntimeException]  
  Aborted                                                 
                                                          

doctrine:migrations:migrate [--write-sql] [--dry-run] [--query-time] [--allow-no-migration] [--configuration [CONFIGURATION]] [--db-configuration [DB-CONFIGURATION]] [--db DB] [--em EM] [--shard SHARD] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env ENV] [--no-debug] [--]  []
```

**Thanks for your time. If I fix the problem I will share with you the solution**

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

**Comment by escopecz** [comment]
Do not Mautic tracking (as a contact) when logged in to Mautic in the same browser. 

Read more about page tracking: https://www.mautic.org/docs/en/pages/troubleshooting.html
Read more about email tracking: https://www.mautic.org/docs/en/emails/troubleshooting.html
Read more about troubleshooting: https://www.mautic.org/docs/en/tips/troubleshooting.html
Read about upgrade issues: https://johnlinhart.com/blog/uh-oh-mautic-upgrade-was-not-successful

Let us know if you won't get answers from any of the links above. I will reopen the issue then.

**Comment by sbolinches** [comment]
Hi escopecz 

I had already read them. The problem is the opposite. It only works when I'm connected.

When a lead sends a form, I receive its data in the CRM (always), but the campaigns only start when I have the session started (logged) in Mautic. But only if I'm connected at this precise moment, because, if not, the campaign for that lead never begins (permanent pending status)

**Comment by escopecz** [comment]
There must be some configuration issue as Mautic was coded the other way around. I'm afraid I can't help as I'm not able reproduce the issue. But I leave it open in case someone else will be.

**Comment by sbolinches** [comment]
Perfect, and thanks for your time.

**Comment by sbolinches** [comment]
@escopecz  I'm sure it's a privilege problem. 
Seems my hosting provider has changed my data base values and has not told me. It's weird, because I used Mautic since version 2.9 and it has always worked well. All of a sudden, since 2.14, weird things happened, like campaign actions were not always carried out. In version 2.15 I realized that they only worked when Mautic is open in the foreground. 

![Privileges](https://user-images.githubusercontent.com/26323868/57532998-02e0d880-733d-11e9-8fdd-ac86636fff1b.jpg)

Thanks for the help
